src/main/java/com/it4045/pokemonviewer/enterprise/service/PokemonServiceStub.java

```diff
@@ -28,6 +28,12 @@ public Pokemon fetchByName(String pokemonName) {
         return pokemon;
     }
 
+    @Override
+    public Pokemon fetchByNumber(int pokemonNumber) {
+        pokemon.setPokemonNumber("3");
+        return pokemon;
+    }
+
```

Comment:
Second technical suggestion: Implemented this functionality in the Pokemon service stub class, retrieving a Pokemon by number.

Here is where I implemented the functionality to retrieve Pokemon by their number. Following the example in which a Pokemon with the name "Charizard" is retrieved, I created this method in which a Pokemon with the number "3" is retrieved. I noticed that the pokemonNumber value in the Pokemon class is a string, so that's how I represented it in the body of the function. I think that one of the other code reviewers noted that this value could be changed to an integer, so if you decide to do that, this could simply be changed to the integer value 3 and it should work as expected.